core_data_modules/traced_data/traced_data.py

```diff
@@ -27,6 +27,17 @@ def __init__(self, user, source, timestamp):
 
     def __eq__(self, other):
         return self.user == other.user and self.source == other.source and self.timestamp == other.timestamp
+    
+    def serialize(self):
+        return {
```

Comment:
I think my underlying concern here is the risk that we add something to the object that doesn't get serialised because it's not in this list and we don't automatically serialise all the properties of the object any more.

Could we consider adding a list of all the expected properties of the objects, and to raise an exception if we see something unexpected? The downside of this is that it may cause issues when the version of Python changes. What do you think?

Reply:
I think LGTM other than this - but I'd like to think about your response before we merge due to the size of the change.

Reply:
Can add a round-trip test (with custom equality check)?

Reply:
That would make me a lot more confident

Reply:
thanks

Reply:
Added here: https://github.com/AfricasVoices/CoreDataModules/blob/14721c43d4eb6eed08f674c289c1456d25e3b9e0/tests/traced_data/test_io.py#L450-L464

Reply:
This UI is completely incomprehensible. What were Github thinking?

Reply:
Is there a commit I could look at for this instead?

Reply:
Oh, never mind, if I click the link, I get something that's sensible

Reply:
@as2388:

- Do you need to delete the tempfile, or does the API do that?
- I think you also need an assert that len(expected) == len(imported)? Perhaps the zip implementation in Python does that implicitly, but I'd rather not rely on it.

Otherwise LGTM thanks!

Reply:
Tempfiles delete automatically (https://docs.python.org/3/library/tempfile.html#tempfile.NamedTemporaryFile).

Added the assert